### PR TITLE
Fix Undefined symbols compile error on macOS.

### PR DESCRIPTION
**Pull request by offroad-bis**
Added block of code defining -undefined dynamic_lookup for a LINK_FLAG to fix Undefined symbols for architecture x86_64: error when compiling.  Code block was taken directly from the CMakeLists.txt file located in the mapscript/php directory.

### CODE REVIEW

**Comment by rouault** [comment]
I'm wondering if a better fix wouldn't be to have SWIG_LINK_LIBRARIES(php_mapscriptng ${PHP_LIBRARY}) to be used on all platforms instead of jut on WIN32 (see line 62 of 
mapscript/phpng/CMakeLists.txt)

**Comment by offroad-bis** [comment]
Unfortunately, I don't know much about linking and compiling. I am just starting to learn so I can hopefully help contribute.  The information I found while researching the errors that I was getting when compiling was all related to setting a link flag of -undefined dynamic_lookup for macOS.  When I found how that had been done for APPLE in php_mapscript CMakeLists.txt file, I tried it in the phpng CMakeLists.txt file and the compile worked.  Here are the errors I was getting.

`[100%] Linking C shared module php_mapscriptng.so
cd /tmp/mapserver-20190822-2120-1mn73yf/mapserver-7.4.1/build/mapscript/phpng && /usr/local/Cellar/cmake/3.15.2/bin/cmake -E cmake_link_script CMakeFiles/php_mapscriptng.dir/link.txt --verbose=1
/usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang  -DACCEPT_USE_OF_DEPRECATED_PROJ_API_H -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk -bundle -Wl,-headerpad_max_install_names  -o php_mapscriptng.so CMakeFiles/php_mapscriptng.dir/CMakeFiles/php_mapscriptng.dir/mapscriptPHP7_wrap.c.o ../../libmapserver.7.4.1.dylib /usr/local/lib/libpng.dylib /usr/lib/libz.dylib /usr/local/lib/libjpeg.dylib /usr/local/lib/libfreetype.dylib /usr/local/lib/libproj.dylib /usr/local/lib/libprotobuf-c.dylib /usr/lib/libiconv.dylib /usr/local/lib/libcairo.dylib /usr/local/lib/libgeos_c.dylib /usr/local/opt/libpq/lib/libpq.dylib /usr/local/lib/libgdal.dylib /usr/local/opt/curl-openssl/lib/libcurl.dylib /usr/local/opt/libxml2/lib/libxml2.dylib /usr/local/lib/libgif.dylib -lm 
Undefined symbols for architecture x86_64:
  "__convert_to_string", referenced from:
      __wrap_CompositingFilter_filter_set in mapscriptPHP7_wrap.c.o
      __wrap_clusterObj_region_set in mapscriptPHP7_wrap.c.o
      __wrap_clusterObj_updateFromString in mapscriptPHP7_wrap.c.o
      __wrap_clusterObj_setGroup in mapscriptPHP7_wrap.c.o
      __wrap_clusterObj_setFilter in mapscriptPHP7_wrap.c.o
      __wrap_outputFormatObj_name_set in mapscriptPHP7_wrap.c.o
      __wrap_outputFormatObj_mimetype_set in mapscriptPHP7_wrap.c.o
      ...
  "__efree", referenced from:
      _SWIG_SetPointerZval in mapscriptPHP7_wrap.c.o
      __wrap_destroy_p_scaleTokenEntryObj in mapscriptPHP7_wrap.c.o
      __swig_default_rsrc_destroy in mapscriptPHP7_wrap.c.o
      __wrap_destroy_p_markerCacheMemberObj in mapscriptPHP7_wrap.c.o
      __wrap_destroy_p_labelCacheMemberObj in mapscriptPHP7_wrap.c.o
      __wrap_destroy_p__LayerCompositer in mapscriptPHP7_wrap.c.o
      __wrap_destroy_p_lineObj in mapscriptPHP7_wrap.c.o
      ...
  "__emalloc", referenced from:
      _SWIG_SetPointerZval in mapscriptPHP7_wrap.c.o
      __wrap_CompositingFilter_filter_get in mapscriptPHP7_wrap.c.o
      __wrap_fontSetObj_filename_get in mapscriptPHP7_wrap.c.o
      __wrap_clusterObj_region_get in mapscriptPHP7_wrap.c.o
      __wrap_clusterObj_convertToString in mapscriptPHP7_wrap.c.o
      __wrap_clusterObj_getGroupString in mapscriptPHP7_wrap.c.o
      __wrap_clusterObj_getFilterString in mapscriptPHP7_wrap.c.o
      ...
  "__emalloc_16", referenced from:
      _SWIG_SetPointerZval in mapscriptPHP7_wrap.c.o
      __wrap_imageObj_getBytes in mapscriptPHP7_wrap.c.o
      __wrap_msIO_getStdoutBufferBytes in mapscriptPHP7_wrap.c.o
  "__emalloc_56", referenced from:
      _SWIG_SetPointerZval in mapscriptPHP7_wrap.c.o
      __wrap_layerObj_utfdata_get in mapscriptPHP7_wrap.c.o
  "__zend_get_parameters_array_ex", referenced from:
      __wrap_swig_mapscript_alter_newobject in mapscriptPHP7_wrap.c.o
      __wrap_swig_mapscript_get_newobject in mapscriptPHP7_wrap.c.o
      __wrap_new_intarray in mapscriptPHP7_wrap.c.o
      __wrap_intarray_getitem in mapscriptPHP7_wrap.c.o
      __wrap_intarray_setitem in mapscriptPHP7_wrap.c.o
      __wrap_intarray_cast in mapscriptPHP7_wrap.c.o
      __wrap_intarray_frompointer in mapscriptPHP7_wrap.c.o
      ...
`

**Comment by rouault** [comment]
> Here are the errors I was getting.

yes, those are symbols from the PHP library, hence my suggestion to link against it. The trick used in mapscript/php feels more like a hack than a proper fix

So basically, I'm suggestiong (untested !)
```
$ git diff ../mapscript/phpng/CMakeLists.txt
diff --git a/mapscript/phpng/CMakeLists.txt b/mapscript/phpng/CMakeLists.txt
index 75593e9..2e78a6b 100644
--- a/mapscript/phpng/CMakeLists.txt
+++ b/mapscript/phpng/CMakeLists.txt
@@ -59,9 +59,10 @@ if(WIN32)
   if(WITH_THREAD_SAFETY)
     target_compile_options(php_mapscriptng PRIVATE /DZTS=1)
   endif(WITH_THREAD_SAFETY)
-  SWIG_LINK_LIBRARIES(php_mapscriptng ${PHP_LIBRARY})
 endif(WIN32)
 
+SWIG_LINK_LIBRARIES(php_mapscriptng ${PHP_LIBRARY})
+
 # hide "defined but not used" warnings
 target_compile_options(php_mapscriptng PRIVATE -Wno-unused-label)
 
```
Can you test this ?

**Comment by offroad-bis** [comment]
I got the following errors when running cmake after making the changes.

bash-3.2$ cmake ..
-- /usr/local/Cellar/php/7.3.8/include/php/main
-- Found PHP-Version 7.3.8 (using /usr/local/bin/php-config)
-- PHP Include Path: /usr/include/php;/usr/local/include/php;/usr/include/php;/usr/local/include/php;/usr/local/apache/php;/usr/local/include/php/Zend;/usr/local/include/php/main;/usr/local/include/php/TSRM;INTERNAL;PHP include paths
-- * Summary of configured options for this build
--  * Mandatory components
--   * png: /usr/local/lib/libpng.dylib
--   * jpeg: /usr/local/lib/libjpeg.dylib
--   * freetype: /usr/local/lib/libfreetype.dylib
--  * Optional components
--   * GDAL: /usr/local/lib/libgdal.dylib
--   * OGR: /usr/local/lib/libgdal.dylib
--   * GIF: /usr/local/lib/libgif.dylib
--   * MYSQL: disabled
--   * FRIBIDI: disabled
--   * HARFBUZZ: disabled
--   * GIF: /usr/local/lib/libgif.dylib
--   * CAIRO: /usr/local/lib/libcairo.dylib
--   * SVGCAIRO: disabled
--   * RSVG: disabled
--   * CURL: /usr/local/opt/curl-openssl/lib/libcurl.dylib
--   * PROJ: /usr/local/lib/libproj.dylib
--   * PIXMAN: disabled
--   * LIBXML2: /usr/local/opt/libxml2/lib/libxml2.dylib
--   * POSTGIS: /usr/local/opt/libpq/lib/libpq.dylib
--   * GEOS: /usr/local/lib/libgeos_c.dylib
--   * FastCGI: /usr/local/lib/libfcgi.dylib
--   * PROTOBUFC: /usr/local/lib/libprotobuf-c.dylib
--   * Oracle Spatial: disabled
--   * Exempi XMP: disabled
--  * Optional features
--   * WMS SERVER: ENABLED
--   * WFS SERVER: ENABLED
--   * WCS SERVER: ENABLED
--   * SOS SERVER: ENABLED
--   * WMS CLIENT: ENABLED
--   * WFS CLIENT: ENABLED
--   * ICONV: ENABLED
--   * Thread-safety support: disabled
--   * KML output: ENABLED
--   * Z+M point coordinate support: disabled
--   * XML Mapfile support: disabled
--  * Mapscripts
--   * Python: disabled
--   * PHP: disabled
--   * PHPNG: ENABLED
--   * PERL: disabled
--   * RUBY: disabled
--   * JAVA: disabled
--   * C#: disabled
--   * V8 Javascript: disabled
--   * Apache Module (Experimental): disabled
-- 
-- Will install files to /usr/local/Cellar/mapserver/7.4.1_1
-- Will install libraries to /usr/local/Cellar/mapserver/7.4.1_1/lib
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
PHP_LIBRARY
    linked by target "php_mapscriptng" in directory /tmp/mapserver-20190822-2120-1mn73yf/mapserver-7.4.1/mapscript/phpng

-- Configuring incomplete, errors occurred!
See also "/tmp/mapserver-20190822-2120-1mn73yf/mapserver-7.4.1/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/mapserver-20190822-2120-1mn73yf/mapserver-7.4.1/build/CMakeFiles/CMakeError.log".





On Aug 22, 2019, at 11:06 AM, Even Rouault <notifications@github.com> wrote:

Here are the errors I was getting.

yes, those are symbols from the PHP library, hence my suggestion to link against it. The trick used in mapscript/php feels more like a hack than a proper fix

So basically, I'm suggestiong (untested !)

$ git diff ../mapscript/phpng/CMakeLists.txt
diff --git a/mapscript/phpng/CMakeLists.txt b/mapscript/phpng/CMakeLists.txt
index 75593e9..2e78a6b 100644
--- a/mapscript/phpng/CMakeLists.txt
+++ b/mapscript/phpng/CMakeLists.txt
@@ -59,9 +59,10 @@ if(WIN32)
   if(WITH_THREAD_SAFETY)
     target_compile_options(php_mapscriptng PRIVATE /DZTS=1)
   endif(WITH_THREAD_SAFETY)
-  SWIG_LINK_LIBRARIES(php_mapscriptng ${PHP_LIBRARY})
 endif(WIN32)
 
+SWIG_LINK_LIBRARIES(php_mapscriptng ${PHP_LIBRARY})
+
 # hide "defined but not used" warnings
 target_compile_options(php_mapscriptng PRIVATE -Wno-unused-label)
 
Can you test this ?

**Comment by rouault** [comment]
OK, then it means that the find_library() in cmake/FindPHP.cmake doesn't manage to find the PHP library on your environment

**Comment by rouault** [comment]
Hum, ok, I'm neither good at CMake or PHP, so I'll merge your patch as it. Thanks
